plothist/plothist_style.py

```diff
@@ -37,6 +41,134 @@ def set_style(style="default"):
+        colors.
+    hue : float, 0 <= hue <= 1, optional
+        Saturation of the colors.
+    darkest : float, 0 <= darkest <= 1, optional
+        Intensity of the darkest color in the palette.
+    lightest : float, 0 <= lightest <= 1, optional
+        Intensity of the lightest color in the palette.
+    reverse : bool, optional
+        If True, the palette will go from dark to light.
+
+    Returns
+    -------
+    list of RGB tuples
+        The generated palette of colors represented as a list of RGB tuples.
+
+
+    References
+    ----------
+    [1] Green, D. A. (2011). "A colour scheme for the display of astronomical
+    intensity images". Bulletin of the Astromical Society of India, Vol. 39,
+    p. 289-295.
+    """
+
+    def f(x0, x1):
+        # Adapted from matplotlib
+        def color(lambda_):
+            # emphasise either low intensity values (gamma < 1),
+            # or high intensity values (γ > 1)
+            lambda_gamma = lambda_ ** gamma
+
+            # Angle and amplitude for the deviation
+            # from the black to white diagonal
+            # in the plane of constant perceived intensity
+            a = hue * lambda_gamma * (1 - lambda_gamma) / 2
+
+            phi = 2 * np.pi * (start / 3 + rotation * lambda_)
+
+            return lambda_gamma + a * (x0 * np.cos(phi) + x1 * np.sin(phi))
+
+        return color
+
+    cdict = {
+        "red": f(-0.14861, 1.78277),
+        "green": f(-0.29227, -0.90649),
+        "blue": f(1.97294, 0.0),
+    }
+
+    cmap = mpl.colors.LinearSegmentedColormap("cubehelix", cdict)
+
+    x = np.linspace(lightest, darkest, int(ncolors))
+    pal = cmap(x)[:, :3].tolist()
+    if reverse:
+        pal = pal[::-1]
+    return pal
+
+
+def get_color_palette(cmap, N):
+    """
+    Get N different colors from a chosen colormap.
+
+    Parameters
+    ----------
+    cmap : str
+        The name of the colormap to use. Use "ggplot" get the cycle of the default style. Use "cubehelix" to get the cubehelix palette with default settings. Can also be any colormap from matplotlib (we recommand "viridis", "coolwarm" or "YlGnBu_r").
+    N : int
+        The number of colors to sample.
+
+    Returns
+    -------
+    list
+        A list of RGB color tuples sampled from the colormap.
+
+    References
+    ----------
+    ggplot colormap: https://matplotlib.org/stable/gallery/style_sheets/ggplot.html
+    Matplotlib colormaps: https://matplotlib.org/stable/gallery/color/colormap_reference.html
+
+    See also
+    --------
+    cubehelix_palette : Make a sequential palette from the cubehelix system.
+    """
+    if N < 2:
+        raise ValueError("The number of colors asked should be >1.")
+
+    if cmap == "ggplot":
+        if N > 7:
+            warnings.warn(
+                f"Only 7 colors are available in the default style cycle ({N} asked). Will return 7 colors.",
+            )
+            N = 7
```

Comment:
No, if the user wants more than 7 colors, we should raise a value error, not a warning.